pkgs/by-name/th/theia-ide/package.nix

```diff
@@ -0,0 +1,208 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+
+  copyDesktopItems,
+  fetchYarnDeps,
+  makeDesktopItem,
+  writableTmpDirAsHomeHook,
+  yarnBuildHook,
+  yarnConfigHook,
+  yarnInstallHook,
+
+  libsecret,
+  xorg,
+
+  electron_37,
+  jdk,
+  nodejs,
+
+  cacert,
+  node-gyp,
+  pkg-config,
+  python3,
+  ripgrep,
+}:
+let
+  electron = electron_37;
+
+  version = "1.64.100";
+
+  src = fetchFromGitHub {
+    owner = "eclipse-theia";
+    repo = "theia-ide";
+    tag = "v${version}";
+    hash = "sha256-kt+J4j48XsPcZa7nUhK2mKaViBx9356gb28G9fYGWxE=";
+  };
+
+  yarnOfflineCache = fetchYarnDeps {
+    yarnLock = src + "/yarn.lock";
+    hash = "sha256-BU2iM1y2IBEBxs82SeGx7h9VYifxFczP/68aWrxWr/A=";
+  };
+
+  theiaPlugins = stdenv.mkDerivation {
+    pname = "theia-plugins";
+    inherit version src yarnOfflineCache;
+
+    nativeBuildInputs = [
+      cacert
+      nodejs
+      writableTmpDirAsHomeHook
+      yarnConfigHook
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      yarn download:plugins
+      mkdir -p $out
+      cp -vr plugins $out
+
+      runHook postInstall
+    '';
+
+    dontPatchShebangs = true;
+
+    outputHash = "sha256-14+V7DREcDuLJl+Z/a2BUE8jNdIPvFgNLgq5/Pz7Rzw=";
```

Comment:
Most likely, this hash differs on darwin because natives are downloaded like a jdk and `jdtls`.